backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CpuPinningPolicyUnit.java

```diff
@@ -74,18 +75,27 @@ public CpuPinningPolicyUnit(PolicyUnit policyUnit,
         switch (vm.getCpuPinningPolicy()) {
             case MANUAL:
             case RESIZE_AND_PIN_NUMA:
-                if (StringUtils.isEmpty(cpuPinning)) {
+                if (StringUtils.isEmpty(cpuPinning) && vm.getCpuPinningPolicy() == CpuPinningPolicy.MANUAL) {
                     // The logic below applies to MigrateVm in which we get here with CPU pinning that was set on the
                     // source host. On RunVm, we get here before determining the CPU pinning for the VM so the dynamic
                     // CPU pinning is not set and hosts should not be filtered by the logic below.
                     return hosts;
                 }
-                // collect all pinned host cpus and merge them into one set
-                final Set<Integer> pinnedCpus = CpuPinningHelper.getAllPinnedPCpus(cpuPinning);
+                Set<Integer> pinnedCpus = new LinkedHashSet<>();
+                if (!StringUtils.isEmpty(cpuPinning)) {
+                    // collect all pinned host cpus and merge them into one set
+                    pinnedCpus = CpuPinningHelper.getAllPinnedPCpus(cpuPinning);
+                }
```

Comment:
what's the reason for this change?

Reply:
if we are just running now `resize and pin numa` vm, the `cpuPinning` is null. When calling `CpuPinningHelper.getAllPinnedPCpus(cpuPinning);` we will hit an exception. we should bypass it (the above comment).